xray/src/bin/merge_xray_quadtrees.rs

```diff
@@ -20,8 +24,154 @@ struct CommandlineArguments {
     input_directories: Vec<PathBuf>,
 }
 
+fn copy_images(input_directory: &Path, output_directory: &Path) {
+    if input_directory == output_directory {
+        return;
+    }
+    globwalk::GlobWalkerBuilder::new(
+        input_directory,
+        &format!("*.{}", xray::IMAGE_FILE_EXTENSION),
+    )
+    .build()
+    .expect("Failed to build GlobWalker")
+    .filter_map(Result::ok)
+    .for_each(|dir_entry| {
+        std::fs::copy(
+            dir_entry.path(),
+            output_directory.join(dir_entry.file_name()),
+        )
+        .expect(&format!(
+            "Failed to copy {:?} to {:?}.",
+            dir_entry.path(),
+            output_directory
+        ));
+    })
+}
+
+fn copy_all_images(input_directories: &[PathBuf], output_directory: &Path) {
+    for input_directory in input_directories {
+        copy_images(input_directory, output_directory);
+    }
+}
+
+fn read_metadata_from_directory(directory: &Path) -> Vec<Meta> {
+    globwalk::GlobWalkerBuilder::new(directory, "meta*.pb")
+        .build()
+        .expect("Failed to build GlobWalker")
+        .filter_map(Result::ok)
+        .map(|dir_entry| PathBuf::from(dir_entry.path()))
+        .map(|path| Meta::from_disk(&path).expect(&format!("Failed to load meta from {:?}.", path)))
```

Comment:
Nit: These two lines can be merged, right?

Reply:
done